msbuild/Xamarin.MacDev.Tasks.Core/VerbosityUtils.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corp
+
+using System;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.MacDev.Tasks {
+
+	// Helper code for Task delegating work to external tools but that still
+	// needs to be verbosity-aware
+	public static class VerbosityUtils {
+
+		// verbosity can be set in multiple ways
+		// this makes it a consistent interpretation of them
+		static public string Merge (string extraArguments, LoggerVerbosity taskVerbosity)
+		{
+			string result = String.Empty;
+			var empty_extra = String.IsNullOrEmpty (extraArguments);
+			// We give the priority to the extra arguments given to the tools
+  			if (empty_extra || (!empty_extra && !extraArguments.Contains ("-q") && !extraArguments.Contains ("-v"))) {
+				// if nothing is specified fall back to msbuild settings
+				// first check if some were supplied on the command-line
+				result = GetVerbosityLevel (Environment.CommandLine);
+				// if not then use the default from the msbuild config files, which Visual Studio for Mac can override (to match the IDE setting for msbuild)
+				if (result.Length == 0)
+					result = GetVerbosityLevel (taskVerbosity);
+			}
+			return result;
+		}
+
+		//
+		// This is an hack, since there can be multiple loggers.
+		// However it's the most common use case and `mtouch` logs
+		// are often the most important to gather and developers expect
+		// a single change (in verbosity) to do the job and be consistent in CI.
+		//
+		// msbuild argument format
+		//	-verbosity:<level> Display this amount of information in the event log.
+		//		The available verbosity levels are: q[uiet], m[inimal],
+		//		n[ormal], d[etailed], and diag[nostic]. (Short form: -v)
+		//
+		static public string GetVerbosityLevel (string commandLine)
+		{
+			var verbosity = String.Empty;
```

Comment:
There's a `StringUtils.TryParseArguments` method you can use to split a command line into separate arguments: https://github.com/xamarin/xamarin-macios/blob/8e2f221063bf2df4e9e5135734e1223790b02ac3/tools/common/StringUtils.cs#L126

I think it could make the code somewhat simpler:

```csharp
if (!StringUtils.TryParseArguments (commandLine, out var args, _))
    return string.Empty;
foreach (var arg in args) {
    if (arg.Length == 0)
        continue;
    if (arg [0] != '/' && arg [0] != '-')
        continue;
    arg = arg.Substring (1);
    if (!arg.StartsWith ("verbosity:", StringComparison.OrdinalIgnoreCase))
        continue;
    var value = arg.Substring ("verbosity:".Length);
    switch (value) {
    case "q":
    case ...:
    }
}
```